### DISCUSSION TITLE
WB crashes VM under Linux

### DISCUSSION THREAD

**Discussion by vogella** [eclipse/windowbuilder#173] [General]
Trying to use WB under Linux crashes the VM.

I tried with Java 11 hotspot, J9 and Java 17 Temurin.

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by net.sf.cglib.core.ReflectUtils$2 (file:/home/vogella/git/windowbuilder/org.eclipse.wb.runtime.lib/lib/cglib-nodep-2.2.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of net.sf.cglib.core.ReflectUtils$2
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Unhandled exception
Type=Segmentation error vmState=0x00040000
J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000002
Handler1=00007FB9B3083380 Handler2=00007FB9B2DE3A10 InaccessibleAddress=00007FB8F663B000
RDI=00007FB8F663B190 RSI=00007F9FF04066E0 RAX=00007FB8F663B010 RBX=0000000000000700
RCX=00000000000006C0 RDX=00007F9FF0406160 R8=00007F9FF0406160 R9=0000000000000039
R10=00007FB8F663ADB0 R11=0000000000000700 R12=0000000000000700 R13=000000000000001B
R14=0000000000000000 R15=0000000000000000
RIP=00007FB9705DCC76 GS=0000 FS=0000 RSP=00007FB9B43A6908
EFlags=0000000000010202 CS=0033 RBP=0000000000000320 ERR=0000000000000004
TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=00007FB8F663B000
xmm0 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm1 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm2 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm3 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm4 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm5 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm6 bf70624dd2f1a9fc (f: 3539053056,000000, d: -4,000000e-03)
xmm7 4072a010624dd2f2 (f: 1649267456,000000, d: 2,980040e+02)
xmm8 3ff0000000000000 (f: 0,000000, d: 1,000000e+00)
xmm9 415fffffc0000000 (f: 3221225472,000000, d: 8,388607e+06)
xmm10 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm11 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm12 0000000000000000 (f: 0,000000, d: 0,000000e+00)
xmm13 40068f5c28f5c28f (f: 687194752,000000, d: 2,820000e+00)
xmm14 c018000000000000 (f: 0,000000, d: -6,000000e+00)
xmm15 3ed18776c5cd3500 (f: 3318560000,000000, d: 4,179276e-06)
Module=/lib/x86_64-linux-gnu/libpixman-1.so.0
Module_base_address=00007FB97054F000
Target=2_90_20210421_975 (Linux 5.13.0-22-generic)
CPU=amd64 (8 logical CPUs) (0x7c51d5000 RAM)
----------- Stack Backtrace -----------
(0x00007FB9705DCC76 [libpixman-1.so.0+0x8dc76])
(0x00007FB9705DCE4C [libpixman-1.so.0+0x8de4c])
pixman_image_composite32+0x43d (0x00007FB97055A21D [libpixman-1.so.0+0xb21d])
(0x00007FB970B603FD [libcairo.so.2+0x333fd])
(0x00007FB970B9861F [libcairo.so.2+0x6b61f])
(0x00007FB970B98DC0 [libcairo.so.2+0x6bdc0])
(0x00007FB970B539F1 [libcairo.so.2+0x269f1])
(0x00007FB970BA5918 [libcairo.so.2+0x78918])
(0x00007FB970B5BFAA [libcairo.so.2+0x2efaa])
cairo_paint+0x1e (0x00007FB970BB376E [libcairo.so.2+0x8676e])
(0x00007FB841873D05 [libwbp3.so+0x1d05])
(0x00007FB841873E10 [libwbp3.so+0x1e10])
(0x00007FB841873ED3 [libwbp3.so+0x1ed3])
(0x00007FB841873F31 [libwbp3.so+0x1f31])
(0x00007FB841873F93 [libwbp3.so+0x1f93])
Java_org_eclipse_wb_internal_os_linux_OSSupportLinux__1makeShot+0xd2 (0x00007FB84187438E [libwbp3.so+0x238e])
(0x00007FB9B31F0926 [libj9vm29.so+0x1ab926])
(0x00007FB9B31EF8C7 [libj9vm29.so+0x1aa8c7])
(0x00007FB9B312CDFC [libj9vm29.so+0xe7dfc])
(0x00007FB9B310C74D [libj9vm29.so+0xc774d])
(0x00007FB9B317F6B2 [libj9vm29.so+0x13a6b2])
---------------------------------------
JVMDUMP039I Speicherauszugsereignis "gpf", Detail "" um 2021/12/21 17:09:41 - bitte warten.
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von System mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/core.20211221.170941.35363.0001.dmp" an
JVMDUMP039I Speicherauszugsereignis "abort", Detail "" um 2021/12/21 17:09:41 - bitte warten.
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von System mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/core.20211221.170941.35363.0002.dmp" an
JVMPORT030W /proc/sys/kernel/core_pattern-Einstellung "|/usr/share/apport/apport %p %s %c %d %P %E" gibt an, dass der Kernspeicherauszug über eine Pipe in ein externes Programm geleitet werden soll.  Versuchte Umbenennung von core oder core.35660.

JVMDUMP012E Fehler in System-Speicherauszug: The core file created by child process with pid = 35660 was not found. Expected to find core file with name "/home/vogella/dev/eclipse-2021-11-10/eclipse/core.35660"
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von Java mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/javacore.20211221.170941.35363.0003.txt" an
JVMDUMP010I Speicherauszug von Java in /home/vogella/dev/eclipse-2021-11-10/eclipse/javacore.20211221.170941.35363.0003.txt geschrieben
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von Snap mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/Snap.20211221.170941.35363.0004.trc" an
JVMDUMP010I Speicherauszug von Snap in /home/vogella/dev/eclipse-2021-11-10/eclipse/Snap.20211221.170941.35363.0004.trc geschrieben
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von JIT mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/jitdump.20211221.170941.35363.0005.dmp" an
JVMDUMP010I Speicherauszug von JIT in /home/vogella/dev/eclipse-2021-11-10/eclipse/jitdump.20211221.170941.35363.0005.dmp geschrieben
JVMDUMP013I Speicherauszugsereignis "abort", Detail "" wurde verarbeitet.
JVMPORT030W /proc/sys/kernel/core_pattern-Einstellung "|/usr/share/apport/apport %p %s %c %d %P %E" gibt an, dass der Kernspeicherauszug über eine Pipe in ein externes Programm geleitet werden soll.  Versuchte Umbenennung von core oder core.35659.

JVMDUMP012E Fehler in System-Speicherauszug: The core file created by child process with pid = 35659 was not found. Expected to find core file with name "/home/vogella/dev/eclipse-2021-11-10/eclipse/core.35659"
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von Java mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/javacore.20211221.170941.35363.0002.txt" an
JVMDUMP010I Speicherauszug von Java in /home/vogella/dev/eclipse-2021-11-10/eclipse/javacore.20211221.170941.35363.0002.txt geschrieben
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von Snap mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/Snap.20211221.170941.35363.0003.trc" an
JVMDUMP010I Speicherauszug von Snap in /home/vogella/dev/eclipse-2021-11-10/eclipse/Snap.20211221.170941.35363.0003.trc geschrieben
JVMDUMP032I JVM forderte als Antwort auf ein Ereignis einen Speicherauszug von JIT mit "/home/vogella/dev/eclipse-2021-11-10/eclipse/jitdump.20211221.170941.35363.0004.dmp" an
JVMDUMP010I Speicherauszug von JIT in /home/vogella/dev/eclipse-2021-11-10/eclipse/jitdump.20211221.170941.35363.0004.dmp geschrieben
JVMDUMP013I Speicherauszugsereignis "gpf", Detail "" wurde verarbeitet.

**Comment by lcaron** [comment]
Hi @vogella 

Can you give more information ?

I was able to "play" with WB with Java 17. Here is what I did with Fedora release 29 + GTK 3.14
1) Install Java 17 (Oracle) 
2) Install Eclipse 2021-12 for Commiters. Eclipse runs with the JDK 17 I've installed before
3) Install WB latest good build (drag n'drop from https://www.eclipse.org/windowbuilder/download.php)
4) Create a new SWT project

Do you want me to try with Open JDK 17 ?
